### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by khrystyna-sabat**
[DEMO LINK](https://khrystyna-sabat.github.io/react_list-of-goods/)

### CODE REVIEW

**Review by Bohdan-Pyschchyk — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -16,60 +15,125 @@ const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
+  none,
+  alphabet,
+  length,
```
Comment (illiarerun) [inline src/App.tsx:20]: Let's use ALLCAPS, cause it's mainly used for such constants values like this
```suggestion
  NONE,
  ALPABET,
  LENGTH,
```

File: src/App.tsx [hunk 2]
```diff
@@ -16,60 +15,125 @@ const goodsFromServer = [
...
+
+    return (
+      <div className="App">
+        {!this.state.isStarted
+        && (
+          <button
+            type="button"
+            onClick={() => {
+              this.setState(state => ({ isStarted: !state.isStarted }));
+            }}
```
Comment (illiarerun) [inline src/App.tsx:77]: LEt's create a function handler for this action

File: src/App.tsx [hunk 3]
```diff
@@ -16,60 +15,125 @@ const goodsFromServer = [
...
+            </button>
+
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({
+                  isReversed: false,
+                  sortType: SortType.none,
+                });
+              }}
```
Comment (illiarerun) [inline src/App.tsx:122]: same here

File: src/App.tsx [hunk 4]
```diff
@@ -16,60 +15,125 @@ const goodsFromServer = [
...
+            >
+              Sort by length
+            </button>
+
+            <button
+              type="button"
+              onClick={() => {
+                this.setState(state => ({
+                  isReversed: !state.isReversed,
+                }));
```
Comment (illiarerun) [inline src/App.tsx:109]: and also here

File: src/App.tsx [hunk 5]
```diff
@@ -16,60 +15,125 @@ const goodsFromServer = [
...
+        )}
+
+        {this.state.isStarted
+        && (
+          <div>
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({ sortType: SortType.alphabet });
+              }}
```
Comment (illiarerun) [inline src/App.tsx:90]: here you can create a handler, smth like a `sortBy` function which will take a `sortType` param with `SortType` type, and just pass needed `SortType`

File: src/App.tsx [hunk 6]
```diff
@@ -16,60 +15,125 @@ const goodsFromServer = [
...
+            >
+              Sort alphabetically
+            </button>
+
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({ sortType: SortType.length });
+              }}
```
Comment (illiarerun) [inline src/App.tsx:99]: and then use it here, also

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 7]
```diff
@@ -17,59 +16,137 @@ const goodsFromServer = [
...
+          <button
+            type="button"
+            onClick={this.started}
+          >
+            Start
+          </button>
+        )}
+
+        {this.state.isStarted
+        && (
```
Comment (jstmpelowycz) [inline src/App.tsx:107]: you'd rather use ternary here

File: src/App.tsx [hunk 8]
```diff
@@ -17,59 +16,137 @@ const goodsFromServer = [
...
-  </div>
-);
+export class App extends React.Component<{}, State> {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  started = () => {
```
Comment (jstmpelowycz) [inline src/App.tsx:62]:
```suggestion
  show = () => {
```

File: src/App.tsx [hunk 9]
```diff
@@ -17,59 +16,137 @@ const goodsFromServer = [
...
+    this.setState(state => ({ isStarted: !state.isStarted }));
+  };
+
+  sortByAlphabetically = () => {
```
Comment (jstmpelowycz) [inline src/App.tsx:66]:
```suggestion
  sortAlphabetically = () => {
```

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**